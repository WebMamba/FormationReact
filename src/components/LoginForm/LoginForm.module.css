.container {
  width: 100%;
  max-width: 350px;
  height: 200px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-direction: column;
  background-color: white;
  border-radius: 14px;
  padding: 8px;
  margin: 16px;
}

.button {
  color: white;
  padding: 16px;
  margin: 8px;
  background-color: #c94288;
  border: 0px;
  border-radius: 14px;
  font-family: "Montserrat", sans-serif;
  font-size: 1em;
  font-weight: 600;
  cursor: pointer;
}
