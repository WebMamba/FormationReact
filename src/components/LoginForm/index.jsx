import { useState } from "react";
import TextField from "../TextField";
import styles from "./LoginForm.module.css";
import PropTypes from "prop-types";

export default function LoginForm(props) {
  const { onLogin } = props;
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();

    onLogin({ email, password });
  };

  return (
    <div className={styles.container}>
      <form onSubmit={handleSubmit}>
        <TextField
          label="Email"
          value={email}
          onChange={(event) => setEmail(event.target.value)}
        />
        <TextField
          label="Mot de passe"
          value={password}
          onChange={(event) => setPassword(event.target.value)}
        />
        <button className={styles.button}>Connexion</button>
      </form>
    </div>
  );
}

LoginForm.propTypes = {
  onLogin: PropTypes.func,
};
