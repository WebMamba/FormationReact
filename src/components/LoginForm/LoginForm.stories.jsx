import LoginForm from ".";

export default {
  title: "Components/LoginForm",
  component: LoginForm,
};

export const Normal = () => <LoginForm />;
