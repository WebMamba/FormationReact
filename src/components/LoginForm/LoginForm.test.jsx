import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LoginForm from ".";

test("render LoginForm without crashing", () => {
  const rendered = render(<LoginForm />);

  expect(rendered.baseElement).toMatchSnapshot();
});

test("execute onLogin when user click on submit button", () => {
  const mockOnLogin = jest.fn();
  render(<LoginForm onLogin={mockOnLogin} />);

  const inputs = screen.getAllByRole("textbox");
  userEvent.type(inputs[0], "test@test.com");
  userEvent.type(inputs[1], "test");

  const button = screen.getByText("Connexion");
  userEvent.click(button);

  expect(mockOnLogin).toHaveBeenCalledTimes(1);
});
