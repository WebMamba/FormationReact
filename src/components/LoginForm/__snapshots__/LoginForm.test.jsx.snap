// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render LoginForm without crashing 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <form>
        <div>
          <input
            class="textbox "
            placeholder="Email"
            type="text"
            value=""
          />
          <p
            class="error_message"
          />
        </div>
        <div>
          <input
            class="textbox "
            placeholder="Mot de passe"
            type="text"
            value=""
          />
          <p
            class="error_message"
          />
        </div>
        <button
          class="button"
        >
          Connexion
        </button>
      </form>
    </div>
  </div>
</body>
`;
